[Frontend] Support image object in llm.chat

sfeng33 opened the pull request:
## Purpose

Fix [17551](https://github.com/vllm-project/vllm/issues/17551)

## Test Plan
Unit test:  
```
pytest tests/entrypoints/test_chat_utils.py
```

Manual test: 
Run a multimodal llm chat with hardcoded image object 
```
python examples/offline_inference/mistral-small.py simple --format hf
```

sfeng33 committed:
add image support

Signed-off-by: sfeng33 <4florafeng@gmail.com>

sfeng33 committed:
Support async

Signed-off-by: sfeng33 <4florafeng@gmail.com>

sfeng33 committed:
Fix test

Signed-off-by: Flora Feng <4florafeng@gmail.com>
Signed-off-by: sfeng33 <4florafeng@gmail.com>

sfeng33 committed:
Format

Signed-off-by: Flora Feng <4florafeng@gmail.com>
Signed-off-by: sfeng33 <4florafeng@gmail.com>

sfeng33 committed:
test mistral

Signed-off-by: sfeng33 <4florafeng@gmail.com>

sfeng33 committed:
Fix precommit and update type

Signed-off-by: sfeng33 <4florafeng@gmail.com>

sfeng33 committed:
Revert "test"

This reverts commit 4faa6cce809a1375b0d380ac67eee76d6e521017.

Signed-off-by: sfeng33 <4florafeng@gmail.com>

sfeng33 committed:
Format

Signed-off-by: Flora Feng <4florafeng@gmail.com>
Signed-off-by: sfeng33 <4florafeng@gmail.com>

DarkLight1337 reviewed (commented):
Can you update the documentation and example files to showcase this way of inputting images?

aarnphm commented on tests/entrypoints/test_chat_utils.py:
```diff
@@ -264,10 +264,8 @@ def test_parse_chat_messages_multiple_images(
-                "image_url": {
-                    "url": image_url
-                }
+                "type": "image",
```
I prefer to have
```suggestion
                "type": "image_pil",
```

given that we add `image_embed`

ywang96 reviewed (approved):
Thanks for working on this and carrying #17919 to the finishing line! The code change is clean to me, but please also address the comments from other reviewers.

sfeng33 committed:
Address feedback

Signed-off-by: sfeng33 <4florafeng@gmail.com>

sfeng33 commented:
Thank you all for the feedback!

Updates in this revision:
* Replaced the type 'image' with 'image_pil'.
* Added a usage example in mistral-small.py.

I didn’t update the documentation since the current docs don’t seem to cover this area. That said, I’m happy to add something if you have suggestions for a good place to include it.

DarkLight1337 commented:
Would be great if you could update the documentation on this page! https://docs.vllm.ai/en/latest/features/multimodal_inputs.html

sfeng33 committed:
Update doc

Signed-off-by: sfeng33 <4florafeng@gmail.com>

sfeng33 commented:
Addressed @DarkLight1337's feedback on documentation.

sfeng33 committed:
Update field name

Signed-off-by: sfeng33 <4florafeng@gmail.com>
